src/lib/io/sol-uart-common.c

```
@@ -0,0 +1,145 @@
+        SOL_STR_TABLE_ITEM("even", SOL_UART_PARITY_EVEN),
+        SOL_STR_TABLE_ITEM("odd", SOL_UART_PARITY_ODD),
+        { }
+    };
+
+    return sol_str_table_lookup_fallback(table,
+        sol_str_slice_from_str(parity), SOL_UART_PARITY_NONE);
+}
+
+SOL_API const char *
+sol_uart_parity_to_str(enum sol_uart_parity parity)
+{
+    static const char *parity_names[] = {
+        [SOL_UART_PARITY_NONE] = "none",
+        [SOL_UART_PARITY_EVEN] = "even",
+        [SOL_UART_PARITY_ODD] = "odd"
+    };
+
+    if (parity < SOL_UTIL_ARRAY_SIZE(parity_names))
+        return parity_names[parity];
+
+    return NULL;
+}
+
+SOL_API enum sol_uart_stop_bits
+sol_uart_stop_bits_from_str(const char *stop_bits)
+{
+    static const struct sol_str_table table[] = {
+        SOL_STR_TABLE_ITEM("stopbits-1", SOL_UART_STOP_BITS_ONE),
+        SOL_STR_TABLE_ITEM("stopbits-2", SOL_UART_STOP_BITS_TWO),
+        { }
+    };
+
+    return sol_str_table_lookup_fallback(table,
+        sol_str_slice_from_str(stop_bits), SOL_UART_STOP_BITS_ONE);
+}
+
+SOL_API const char *
+sol_uart_stop_bits_to_str(enum sol_uart_stop_bits stop_bits)
+{
+    static const char *stop_bits_names[] = {
+        [SOL_UART_STOP_BITS_ONE] = "stopbits-1",
+        [SOL_UART_STOP_BITS_TWO] = "stopbits-2"
+    };
+
+    if (stop_bits < SOL_UTIL_ARRAY_SIZE(stop_bits_names))
+        return stop_bits_names[stop_bits];
+
+    return NULL;
+}
+
+
+SOL_API enum sol_uart_data_bits
+sol_uart_data_bits_from_str(const char *data_bits)
+{
+    static const struct sol_str_table table[] = {
+        SOL_STR_TABLE_ITEM("databits-8", SOL_UART_DATA_BITS_8),
+        SOL_STR_TABLE_ITEM("databits-7", SOL_UART_DATA_BITS_7),
+        SOL_STR_TABLE_ITEM("databits-6", SOL_UART_DATA_BITS_6),
+        SOL_STR_TABLE_ITEM("databits-5", SOL_UART_DATA_BITS_5),
+        { }
+    };
+
+    return sol_str_table_lookup_fallback(table,
+        sol_str_slice_from_str(data_bits), SOL_UART_DATA_BITS_8);
+}
```

@barbieri I understand that it makes sense for this to return ```SOL_UART_DATA_BITS_8``` as a fallback, but that can cause unexpected behaviour the cause for which is hard to trace. Like, in this case, what if somebody looks up ```"dataBits:8"``` or even ```"dataBits-8"```, or even accidentally ```""```? In Javascript, you could throw an exception, making it easy to debug the application.

If we keep this, the implementation of ```sol_str_table_lookup_fallback()``` should at least issue a warning when it falls back.